test/xpu_api/utilities/pairs/pairs.pair/noexcept_swap_g.pass.cpp

```diff
@@ -0,0 +1,65 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/utility>
+
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct TestStruct
+{
+};
+
+void
+kernel_test()
+{
+    sycl::queue deviceQueue = TestUtils::get_test_queue();
+    {
+        deviceQueue.submit([&](sycl::handler& cgh) {
+            cgh.single_task<class KernelTest>([=]() {
+                typedef dpl::pair<int, int> tt1;
+                typedef dpl::pair<int, float> tt2;
+                typedef dpl::pair<short, TestStruct> tt3;
+                typedef dpl::pair<short, TestStruct> tt4;
+                typedef dpl::pair<TestStruct, TestStruct> tt5;
+                typedef dpl::pair<TestStruct, TestStruct> tt6;
+                typedef dpl::pair<TestStruct, float> tt7;
+                typedef dpl::pair<TestStruct, TestStruct> tt8;
```

Comment:
We need a test with the struct that is not nothrow move constructible/assignable. The pair should not be nothrow swappable in this case.

Reply:
Done